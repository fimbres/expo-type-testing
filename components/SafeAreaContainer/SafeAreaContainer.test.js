import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react-native";

import { SafeAreaContainer } from ".";

describe("<SafeAreaContainer />", () => {
  it("render children correctly", () => {
    render(
      <SafeAreaContainer>
        <Text>This is a test</Text>
      </SafeAreaContainer>
    );

    const primaryText = screen.getByText("This is a test");

    expect(primaryText).toBeDefined();
  });

  it("render correctly", () => {
    const tree = renderer
      .create(
        <SafeAreaContainer>
          <Text>This is a test</Text>
        </SafeAreaContainer>
      )
      .toJSON();
    expect(tree.children.length).toBe(1);
    expect(tree).toMatchSnapshot();
  });
});
