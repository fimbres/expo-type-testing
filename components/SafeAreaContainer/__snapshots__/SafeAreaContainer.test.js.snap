// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SafeAreaContainer /> render correctly 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "backgroundColor": "#F8F8F8",
        "flex": 1,
        "paddingTop": 0,
      },
      undefined,
    ]
  }
>
  <Text>
    This is a test
  </Text>
</RCTSafeAreaView>
`;
